### PR TITLE
Ensure TxParamSetupReq is scheduled when necessary

### LINKED ISSUE #690: Flaky NS test
#### Summary

NS test is flaky.

#### Steps to Reproduce

Run CI

#### What do you see now?

https://gist.github.com/johanstokking/45bde10bb3bf6f8636eea7972c4beade

#### What do you want to see instead?

Successful NS test.

### LINKED ISSUE #709: Dwell time requirements should be sent to devices on join
#### Summary
Devices should be sent Dwell Time requirements at time of Join.
Presumably this would be done using MAC command TxParamSetupReq.

This should be done whether the band requires Dwell or not, since the network server cannot know whether the device has it enabled or disabled by default.

#### Steps to Reproduce
1. Register and configure any device for LoRaWAN 1.0.2, OTAA, Class A. I use bandplan AS_920_923, but any band plan will probably work to reproduce.
2. Monitor event names for your app: ttn-lw-cli events --application-id yourAppID | jq .name
3. Issue a join request from your device
4. Note that in the event list, there is no TxParamSetupReq command
5. Note that whether your device is or isn't configured with a Dwell time restriction, there is no change after the join.

#### What do you see now?
Devices that enable dwell by default remain unchanged after the join.
Devices that disable dwell by default remain unchanged after the join.

#### What do you want to see instead?
Devices change their dwell setting on join to be in line with the band plan's dwell requirement, as a result of the network server sending them a TxParamSetupReq command

#### How do you propose to implement this?
Additional MAC command issued at time of Join

#### Can you do this yourself and submit a Pull Request?
Unfortunately not

### PR DESCRIPTION
**Pull request by rvolosatovs**
#### Summary

Closes #709 
Closes #690 

#### Changes

- Send `TxParamSetupReq` when variable(either on or off) dwell time requirement exists
- Fix `TxParamSetupSupport` value in AU915 bands

#### Notes for Reviewers

@johanstokking what do you think about this approach? Essentially `nil` for dwell-time field means that the value is unknown or irrelevant.
When the dwell-time field value is non-nil in desired MAC parameters, it means that both dwell-time options are possible in the band and the value denotes the actual "desired" value(from frequency plan), otherwise we assume the device is compliant with the regional parameters spec in this regard.(in those cases we cannot change it anyway, as `TxParamSetupReq` should not be implemented in those regions according to the spec)

#### Release Notes

- Fixed dwell-time issues in AS923 and AU915 bands.

### CODE REVIEW

File: api/end_device.proto [hunk 1]
```diff
@@ -56,13 +56,10 @@ message Session {
 // This is used internally by the Network Server and is read only.
 message MACParameters {
   option (gogoproto.populate) = false;
+  reserved 2, 3;
 
   // Maximum EIRP (dBm).
   float max_eirp = 1 [(gogoproto.customname) = "MaxEIRP"];
-  // Uplink dwell time is set (400ms).
-  bool uplink_dwell_time = 2;
```
Comment (rvolosatovs) [inline api/end_device.proto]: Is it OK to deprecate these like this or should I add logic, which would handle the migration to the new format?

**Comment by rvolosatovs** [comment]
Blocked by #732

File: api/end_device.proto [hunk 2]
```diff
@@ -110,6 +107,11 @@ message MACParameters {
   }
   // Configured uplink channels and optionally Rx1 frequency.
   repeated Channel channels = 19 [(validate.rules).repeated.min_items = 1];
+
+  // Uplink dwell time is set (400ms).
+  google.protobuf.BoolValue uplink_dwell_time = 20;
+  // Downlink dwell time is set (400ms).
+  google.protobuf.BoolValue downlink_dwell_time = 21;
```
[the remark this answers is not in the document]
Comment (htdvisser) [reply api/end_device.proto]: +1 on @johanstokking's second comment. This is not an optional thing per device, it fully depends on the frequency plan. This isn't an optional thing in the current/desired parameters either.

[the remark this answers is not in the document]
Comment (rvolosatovs) [reply api/end_device.proto]: Dwell-time limitations are indeed driven by the frequency plan and are not a per-device setting. How are these 2 fields related?
`MACState` records the state the device is in. Please read the PR description for additional context.

**Comment by rvolosatovs** [comment]
@johanstokking This is fully driven by frequency plan - there are no per-device settings.

**Comment by rvolosatovs** [comment]
AU915 is indeed missing the `TxParamSetupReqSupport`, I will fix that

**Comment by johanstokking** [comment]
How about AS923? It should always be sent there.

**Comment by rvolosatovs** [comment]
For AS923 the value is correct https://github.com/TheThingsNetwork/lorawan-stack/blob/48662cc8fb3201261c9d7ebc1b714ce337a38dae/pkg/band/as_923.go#L152

**Pull request merged**